core/market/src/market.rs

```diff
@@ -234,6 +235,41 @@ impl MarketService {
         Ok(())
     }
 
+    pub async fn list_agreements(
+        &self,
+        id: &Identity,
+        state: Option<AgreementState>,
+        before: Option<DateTime<Utc>>,
+        after: Option<DateTime<Utc>>,
+        app_sesssion_id: Option<String>,
+    ) -> Result<Vec<AgreementListEntry>, AgreementError> {
+        let agreements = self
+            .db
+            .as_dao::<AgreementDao>()
+            .list(Some(id.identity), state, before, after, app_sesssion_id)
+            .await
+            .map_err(|e| AgreementError::Internal(format!("{:?}", e)))?;
```

Comment:
Could work without debug formatting?

Reply:
Now that I think about it, `e.to_string()` is better.